smd/main.go

```diff
@@ -22,27 +22,52 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"log"
+	"os"
 
 	"sigs.k8s.io/structured-merge-diff/typed"
 )
 
+var (
+	ErrTooManyOperations = errors.New("exactly one of --merge, --compare, or --validate must be provided")
```

Comment:
Just my opinion, but you could have created sub-commands rather than options

Reply:
Yeah, that starts to sound like work, this was just to make it easy to explore with files...